x/auth/ante/basic.go

```diff
@@ -139,3 +145,23 @@ func (cgts ConsumeTxSizeGasDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, sim
 
 	return next(ctx, tx, simulate)
 }
+
+// isIncompleteSignature tests whether SignatureData is fully filled in for simulation purposes
+func isIncompleteSignature(data signing.SignatureData) bool {
+	if data == nil {
+		return true
+	}
+
+	switch data := data.(type) {
+	case *signing.SingleSignatureData:
+		return len(data.Signature) == 0
+	case *signing.MultiSignatureData:
```

Comment:
this case is not tested 👍

Reply:
Added [here](https://github.com/cosmos/cosmos-sdk/pull/6550/files#diff-7d5209e378836fa4fcecb321d23ec5acR112)